src/test/librbd/test_mock_fixture.h

```diff
@@ -25,7 +25,11 @@ ACTION_P(CopyInBufferlist, str) {
 }
 
 ACTION_P2(CompleteContext, r, wq) {
+#if defined(__clang__)
+  ContextWQ *context_wq = static_cast<ContextWQ *>(wq);
```

Comment:
if the libc++'s `stddef.h` looks sane. maybe we should pass to `-stdlib=libc++` the compiler on FreeBSD to use the "new" c++11 stack?